scripts/install-nix-from-closure.sh

```diff
@@ -1,181 +1,700 @@
+        subscribeChannelsRevert() {
+            errorRevert "Received Error signal from line $1"
+            errorRevert "Reverting channels subscriptions: nixpkgs"
+            $nix/bin/nix-channel --remove nixpkgs
+            error "
+
+    Changes reverted.
+    " "$2"
+        }
+        channelUrl='https://nixos.org/channels/nixpkgs-unstable'
+        print "Subscribing to channel: nixpkgs="
+        $nix/bin/nix-channel --add "$channelUrl"
+    }
+
+    updateRepos() {
+        # Subscribe the user to the Nixpkgs channel and fetch it.
+        trap 'updateReposRevert $LINENO $?' INT TERM ABRT QUIT
+        updateReposRevert() {
+            error "
+
+    Received Error signal from line $1
+    " "$2"
+        }
+        print "Pulling information from channel: nixpkgs"
+        # nix-channel should be fully transactional, so no 'nix-channel --rollback'
+        $nix/bin/nix-channel --update nixpkgs
+    }
+
+    notice "
+
+    Performing a single-user installation of Nix:
+    "
+
+    if [ ! -d "$dest" ] ; then
+        createDest
+    fi
 
-Installation finished!  To ensure that the necessary environment
-variables are set, please add the line
+    createStore
 
-  . $p
+    copyNix
 
-to your shell profile (e.g. ~/.profile).
-EOF
-else
-    cat >&2 <<EOF
+    initializeDB
 
-Installation finished!  To ensure that the necessary environment
-variables are set, either log in again, or type
+    setupProfile
 
-  . $p
 
-in your shell.
-EOF
-fi
+    if [ -z "$NIX_SSL_CERT_FILE" ] || [ ! -f "$NIX_SSL_CERT_FILE" ]; then
+        installCerts
+    fi
+
+    if ! $nix/bin/nix-channel --list | grep -q "^nixpkgs "; then
+        subscribeChannels
+    fi
+
+    if [ -z "$_NIX_INSTALLER_TEST" ]; then
+        updateRepos
+    fi
+
+    notice "
+
+    Installation finished. 8)
+    "
+}
+
+postinstall() {
+    notice "
+
+    Launching postinstall tasks:
+    "
+    added=
+    if [ -z "$NIX_INSTALLER_NO_MODIFY_PROFILE" ]; then
+        # Make the shell source nix.sh during login.
+        p=$HOME/.nix-profile/etc/profile.d/nix.sh
+        for i in .bash_profile .bash_login .profile; do
+            fn="$HOME/$i"
+            if [ -w "$fn" ]; then
+                if ! grep -q "$p" "$fn"; then
+                    echo "modifying $fn..." >&2
+                    echo "if [ -e $p ]; then . $p; fi # added by Nix installer"\
+                         >> "$fn"
+                fi
+                added=1
+                break
+            fi
+        done
+    fi
+
+    if [ -z "$added" ]; then
+        notice "
+
+    Installation finished!
+    To ensure that the necessary environment variables are set,
+    please add the line
+
+    . $p
+
+    to your shell profile (e.g. ~/.profile).
+    "
+
+    else
+        notice "
+
+    Installation finished!
+    To ensure that necessary environment variables to use Nix binaries
+    are set in current runtime environment, either log in again, or type
+
+    . $p
```

Comment:
Probably also needs quotes.

Reply:
I don't understand this suggestion.
Do you mean:
```shell
        notice "

    Installation finished!
    To ensure that necessary environment variables to use Nix binaries
    are set in current runtime environment, either log in again, or type

    '. $p'

    in your shell.
    "
```

Without quotes it seems logical, as it matches real life.
```shell

        notice "

    Installation finished!
    To ensure that necessary environment variables to use Nix binaries
    are set in current runtime environment, either log in again, or type

    . $p

    in your shell.
    "
```